### ISSUE TITLE
configuration key not expected error in server,totp and webauthn

### ISSUE THREAD

**Issue by harishdesetti1206** [authelia/chartrepo#146]
"Configuration: configuration key not expected: server.headers.csp_template"
"Configuration: configuration key not expected: totp.disable"
"Configuration: configuration key not expected: webauthn.disable"

The above errors are found in authelia pod logs. The TOTP and WEBAUTHN are disabled in values.yaml but still these errors are displayed.
Even when server.headers.csp_template is left empty, the error arises.

**Comment by james-d-elliott** [comment]
Can't replicate sorry,unless you're running a chart with another version of Authelia this can't happen. If you want to do a semver check to hide those sections we can support that, but we're not using those options in the way you've implemented it.

**Comment by harishdesetti1206** [comment]
@james-d-elliott I'm using latest tag version: 0.8.15.
I have set the configMap.totp.disable true and configMap.webauthn.disable true. The i faced this error.  Not sure if we can put a check to not throw this error.
Also, i will try to create a semver check as you mentioned and raise a MR.

**Comment by harishdesetti1206** [comment]
@james-d-elliott I'm using latest tag version: 0.8.15.
I have set the configMap.totp.disable true and configMap.webauthn.disable true. The i faced this error.  Not sure if we can put a check to not throw this error.
Also, i will try to create a semver check as you mentioned and raise a MR.

**Comment by james-d-elliott** [comment]
What version of Authelia though? This error can't happen on that version of the chart unless you're using differing version.

**Comment by james-d-elliott** [comment]
See this here as an example: 

https://github.com/authelia/chartrepo/blob/master/charts/authelia/templates/configMap.yaml#L225-L228

The likely check you need to do is for the disable option and all of webauthn, it shouldn't be included pre 4.34.0 I believe. So something like this:

```
    totp:
      {{- if (semverCompare ">=4.34.0" (.Values.versionOverride | default .Chart.AppVersion | toString)) }}
      disable: {{ .Values.configMap.totp.disable | default false }}
      {{- end }}
      issuer: {{ default .Values.domain .Values.configMap.totp.issuer }}
      algorithm: {{ default "sha1" .Values.configMap.totp.algorithm }}
      digits: {{ default 6 .Values.configMap.totp.digits }}
      period: {{ default 30 .Values.configMap.totp.period }}
      skew: {{ default 1 .Values.configMap.totp.skew }}
   {{- if (semverCompare ">=4.34.0" (.Values.versionOverride | default .Chart.AppVersion | toString)) }}
    webauthn:
      disable: {{ .Values.configMap.webauthn.disable | default false }}
      display_name: {{ .Values.configMap.webauthn.display_name | default "Authelia" }}
      attestation_conveyance_preference: {{ .Values.configMap.webauthn.attestation_conveyance_preference | default "indirect" }}
      user_verification: {{ .Values.configMap.webauthn.user_verification | default "preferred" }}
      timeout: {{ .Values.configMap.webauthn.timeout | default "60s" }}
      {{- end }}
```

**Comment by harishdesetti1206** [comment]
@james-d-elliott The issue is solved from 4.34.0.
I'm closing this issue. Thanks alot for the support.
